dataflow/src/main/java/org/checkerframework/dataflow/busyexpression/BusyExprValue.java

```diff
@@ -0,0 +1,49 @@
+package org.checkerframework.dataflow.busyexpression;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.AbstractValue;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.javacutil.BugInCF;
+
+/** A busy expression wrapper turning a node into the abstract value */
+public class BusyExprValue implements AbstractValue<BusyExprValue> {
+
+    /**
+     * A busy expression is represented by a node, which can be a {@link
+     * org.checkerframework.dataflow.cfg.node.BinaryOperationNode}
+     */
+    protected final Node busyExpression;
```

Comment:
Yes, fixed it

Reply:
foo()+bar() // what if bar has a side effect

setup a breakpoint in methodinvocation.

Reply:
It looks like using `equals` is faulty. Suppose we have 2 expressions
`b + foo()` both in the `if` and `else` branches, where calling `foo()`
returns different values each time. It seems that our program will analyze
the 2 expressions as the same and make it as the busy expression at the
conditional block. However, foo() is a method invocation, and calling
it will introduce an exceptional block follows it. As the result,
there are at least 2 blocks following foo() block, one with a transfer
result `b + foo()`, another is the exception block with empty transfer
result. Their intersection is empty when propagating to the foo() block.
The abstract value `b+foo()` is lost during backward propagating, hence
it's not a problem in this case.